### ISSUE TITLE
tab/backspace leaves character on screen

### ISSUE THREAD

**Issue by JeanMertz** [jwilm/alacritty#2026]
This one is a bit hard to explain, and I haven't dug too deep yet, but wanted to report the issue.

This is on macOS 10.14.2, using Alacritty, both latest stable (26) and master.

When I tab-complete a command or path, it's as if all characters somehow shift. 

Here's what happens in this case:

* I type `cd ~/Deve`
* I tab complete to `cd ~/Development/`
* Next I tab complete again to show the list of available subpaths, which ends up as `cd ~/Development/ ` (note that there's an extra space inserted at the end, which is unusual)
* If I now backspace to `cd ~/Deve` it shows my cursor after the `e`,
* But if I then tab-complete again, it completes to `cd ~/Deveelopment/` (note the double e)
* Even though the path shown on screen is incorrect, it's only the visual representation that's wrong, because hitting enter actually goes to the `Development` path.

Interestingly enough, the same behaviour does not show up when using Bash (I use Fish as my shell), but perhaps it would trigger with some other commands in Bash, I'm unsure. If I use Fish with Terminal.app however, the behaviour does not appear there, so it does appear to be related to Alacritty somehow.

Below is a Gif showing this in action. You can also see that after the bug appears, if I remove all characters from the line, it keeps the first character visible (i.e. I can't go more to the left with my cursor). But, the character is not actually used in the eventual command, as visually it looks like I type `cls`, but in fact `ls` is executed.

![ezgif com-video-to-gif-2](https://user-images.githubusercontent.com/383250/51443885-203b5980-1cf0-11e9-8fa4-09f8211a00b8.gif)

**Comment by chrisduerr** [comment]
Can this be reproduced with https://github.com/jwilm/alacritty/commit/21f888ec4149e3078830dfcf9a17adda0ec7623a?

**Comment by chrisduerr** [comment]
It's either the commit right after that or https://github.com/jwilm/alacritty/pull/1994.

**Comment by JeanMertz** [comment]
I think I spoke too soon. I did a git checkout of `21f888e` and a `make app`, and then `open target/release/osx/Alacritty.app`, which didn't show the issue.

But now that I do the same for `master`, it also doesn't show the issue.

The active terminal that I'm doing this from clearly still has the issue (`➜ ttarget/release/alacritty`, see the double t).

I'm going to see how I can get this to either show up again when I build and open that version, or if I can get to _not_ show up again if I build master again and start using that version as my main version.

**Comment by JeanMertz** [comment]
Hmm okay, it could have something to do with a specific character being shown on screen actually.

I can now reproduce it when I move a process to the background, which then updates my prompt to show the `✦` character (you can also see this in my Gif above).

If that character is present, the issue can be reproduced, if it isn't (after I close all background processes), the issue is no longer present.

I'll bisect again, with this knowledge, and see what version actually works (if any?)

**Comment by JeanMertz** [comment]
I reverted back to v1.1 (`865727c`), and it too shows the issue, so I doubt there's any point in going back further, and I suspect this never worked? Perhaps with this new knowledge, you might know if this is actually a known issue?

**Comment by chrisduerr** [comment]
I'm glad that the commit still shows that problem, I was a bit afraid of another issue with tabs.

Linking it to specific unicode characters makes a ton of sense, I believe we've had similar issues before. I think there might even be an issue that directly matches this one and the temporary solution would be to just remove the characters in question.

I can't find the other issue, but the underlying issue is https://github.com/jwilm/alacritty/issues/1295, thus this is a duplicate. This should be resolved once that is merged so you can follow that issue to stay up to date.
